vexflow/vexflow.d.ts

```diff
@@ -0,0 +1,1325 @@
+            getCategory() : string;
+            makeSpacer(padding : number) : {getContext: Function, setStave: Function, renderToStave: Function, getMetrics: Function};
+            placeGlyphOnLine(glyph : Glyph, stave : Stave, line : number) : void;
+            setPadding(padding : number) : void;
+            addToStave(stave : Stave, firstGlyph : boolean) : StaveModifier;
+            addToStaveEnd(stave : Stave, firstGlyph : boolean) : StaveModifier;
+            addModifier() : void;
+            addEndModifier() : void;
+        }
+        
+        class StaveNote extends StemmableNote {
+            //TODO remove the following lines once TypeScript allows subclass overrides with type changes and/or inconsistencies mentioned below are fixed
+            buildStem() : StemmableNote;            
+            setStave(stave : Stave) : Note;
+            addModifier(modifier : Modifier, index? : number) : Note;
+            getModifierStartXY() : {x : number, y : number};
+            getDots() : number;
+
+            constructor(note_struct : {type : string, dots? : number, duration : string, clef : string, keys : string[], octave_shift? : number});
+            static CATEGORY : string;
+            static DEBUG : boolean;
+            static STEM_UP : number;
+            static STEM_DOWN : number;
+            static format(notes : StaveNote[] , state : {left_shift : number, right_shift : number, text_line : number}) : boolean;
+            static formatByY(notes : StaveNote[] , state : {left_shift : number, right_shift : number, text_line : number}) : void;
+            static postFormat(notes : StaveNote[]) : boolean;
+            buildStem() : void; //inconsistent type: void -> StaveNote
+            buildNoteHeads() : void;
+            autoStem() : void;
+            calculateKeyProps() : void;
+            getBoundingBox() : BoundingBox;
+            getLineNumber() : number;
+            isRest() : boolean;
+            isChord() : boolean;
+            hasStem() : boolean;
+            getYForTopText(text_line : number) : number;
+            getYForBottomText(text_line : number) : number;
+            setStave(stave : Stave) : StaveNote;
+            getKeys() : string[];
+            getKeyProps() : {key : string, octave : number, line : number, int_value : number, accidental : string, code : number, stroke : number, shift_right : number, displaced : boolean}[];
+            isDisplaced() : boolean;
+            setNoteDisplaced(displaced : boolean) : StaveNote;
+            getTieRightX() : number;
+            getTieLeftX() : number;
+            getLineForRest() : number;
+            getModifierStartXY(position : Modifier.Position, index : number) : {x : number, y : number};
+            setStyle(style : {shadowColor? : string, shadowBlur? : string, fillStyle? : string, strokeStyle? : string}) : void; // inconsistent type: void -> StaveNote
+            setKeyStyle(index : number, style : string) : StaveNote;
```

Comment:
The type of `style` in `StaveNote.setKeyStyle` appears to be an object, not a string, of the same structure as used for `style` in `StaveNote.setStyle`. (also mentioned on http://www.vexflow.com/docs/stavenote.html)